token/client/src/token.rs

```diff
@@ -3846,4 +3846,76 @@ where
         )
         .await
     }
+
+    /// Initialize token-group on a mint
+    pub async fn token_group_initialize<S: Signers>(
+        &self,
+        mint_authority: &Pubkey,
+        update_authority: &Pubkey,
+        max_size: u32,
+        signing_keypairs: &S,
+    ) -> TokenResult<T::Output> {
+        self.process_ixs(
+            &[spl_token_group_interface::instruction::initialize_group(
+                &self.program_id,
+                &self.pubkey,
+                &self.pubkey,
+                mint_authority,
+                Some(*update_authority),
+                max_size,
+            )],
+            signing_keypairs,
+        )
+        .await
+    }
+
+    async fn get_additional_rent_for_new_group(&self) -> TokenResult<u64> {
+        let account = self.get_account(self.pubkey).await?;
+        let account_lamports = account.lamports;
+        let mint_state = self.unpack_mint_info(account)?;
+        let new_account_len = mint_state
+            .try_get_account_len()?
+            .checked_add(ExtensionType::try_calculate_account_len::<Mint>(&[
+                ExtensionType::TokenGroup,
+            ])?)
+            .ok_or(TokenError::Program(
+                spl_token_2022::error::TokenError::Overflow.into(),
+            ))?;
```

Comment:
Is this really the way to do this for fixed-length types or am I missing something?